src/odemis/gui/cont/views.py

```diff
@@ -317,6 +370,32 @@ def on_stage_pos_change(self, p_pos):
         stage_history.append((p_center, p_size))
         self._data_model.stage_history.value = stage_history
 
+        # Update overview image
+        self._update_ovv()
+
+    def _on_current_stream(self, streams):
+        """
+        Called when some VAs affecting the current stream change
+        """
+        # Try to get the current stream
+        try:
+            self.curr_s = streams[0]
+        except IndexError:
+            self.curr_s = None
+
+        if self.curr_s:
+            self._update_ovv()
+            self.curr_s.should_update.subscribe(self._on_stream_update, init=True)
+            self.curr_s.image.subscribe(self._onNewImage)
+
+    def _on_stream_update(self, _):
+        self._update_ovv()
+
+    @wxlimit_invocation(1)  # max 1 Hz
```

Comment:
Just use `@limit_invocation()`, the `wx` variant ensures also that you're running in the main GUI thread, but it's not needed here.